cdap-ui/app/cdap/services/ThemeHelper.ts

```diff
@@ -0,0 +1,130 @@
+ * the License.
+*/
+
+import cssVars from 'css-vars-ponyfill';
+import {objectQuery, isNilOrEmpty} from 'services/helpers';
+
+declare global {
+  interface Window {
+    CDAP_UI_THEME: any;
+  }
+}
+
+export function applyTheme() {
+  if (
+    !objectQuery(window, 'CDAP_UI_THEME', 'styles') ||
+    isNilOrEmpty(window.CDAP_UI_THEME.styles)
+  ) {
+    // need to run this at least once even if there's no theme customization
+    // so that css variables are parsed correctly even in older browsers
+    cssVars();
+    return;
+  }
+
+  // this is what's going on under the hood for modern browsers:
+  // document.documentElement.style.setProperty(`--${cssVar}`, cssValue);
+  cssVars({
+    variables: window.CDAP_UI_THEME.styles,
+  });
+}
+
+interface ThemeObj {
+  [key: string]: any;
+}
+
+function getTheme(): ThemeObj {
+  if (isNilOrEmpty(window.CDAP_UI_THEME)) {
+    return {};
+  }
+
+  let theme: ThemeObj = {};
+  const themeJSON = window.CDAP_UI_THEME;
+  const specVersion = themeJSON['spec-version'];
+
+  if (specVersion === '1.0') {
+    theme = get10Content(theme, themeJSON);
+    theme = get10Features(theme, themeJSON);
+  }
+  return theme;
+}
+
+function get10Content(theme, themeJSON): ThemeObj {
+  const content = themeJSON.content;
+  if (isNilOrEmpty(content)) {
+    return theme;
+  }
+  if ('footer-text' in content) {
+    theme.footerText = content['footer-text'];
+  }
+  if ('footer-link' in content) {
+    theme.footerLink = content['footer-link'];
+  }
+  if ('logo' in content) {
+    const logo = window.CDAP_UI_THEME.content.logo;
+    if (logo.type) {
+      const logoType = logo.type;
+      if (logoType === 'inline') {
+        theme.logo = objectQuery(logo, 'arguments', 'data');
+      } else if (logoType === 'link') {
+        theme.logo = objectQuery(logo, 'arguments', 'url');
+      }
+    }
+  }
+  return theme;
+}
+
+function get10Features(theme, themeJSON): ThemeObj {
+  const features = themeJSON.features;
+  if (isNilOrEmpty(features)) {
+    return theme;
+  }
+  if ('dashboard' in features) {
+    theme.showDashboard = features.dashboard;
+  }
+  if ('reports' in features) {
+    theme.showReports = features.reports;
+  }
+  if ('data-prep' in features) {
+    theme.showDataPrep = features['data-prep'];
+  }
+  if ('pipelines' in features) {
+    theme.showPipelines = features.pipelines;
+  }
+  if ('analytics' in features) {
+    theme.showAnalytics = features.analytics;
+  }
+  if ('rules-engine' in features) {
+    theme.showRulesEngine = features['rules-engine'];
+  }
+  if ('metadata' in features) {
+    theme.showMetadata = features.metadata;
+  }
+  if ('hub' in features) {
+    theme.showHub = features.hub;
+  }
+  if ('footer' in features) {
+    theme.showFooter = features.footer;
+  }
+  if ('ingest-data' in features) {
+    theme.showIngestData = features['ingest-data'];
+  }
+  if ('add-namespace' in features) {
+    theme.showAddNamespace = features['add-namespace'];
```

Comment:
What do you think of defaulting all of these to true? I don't really have a strong opinion, but it might be better for the global object to only have 2 possible values: `true` or `false`. Then we don't need to worry about `null` or `undefined` values.

Reply:
I just think that this would add more work when we add a property to the theme json files. Currently even if these values are `null` or `undefined`, we already have a default fallback in the component's code.